package.json

```diff
@@ -67,7 +67,8 @@
     "sass-loader": "^10.1.0",
     "typescript": "^4.0.3",
     "webpack": "^5.36.2",
-    "webpack-cli": "^4.7.0"
+    "webpack-cli": "^4.7.0",
+    "d3-selection": "^3.0.0"
```

Comment:
seems like `@carbon/charts` requires this to run okay, typescript complains as shown below
![Screenshot 2021-08-10 at 11 37 15](https://user-images.githubusercontent.com/28008754/128836046-e0522021-d43b-47b7-b396-c15b58ae43eb.png)

Reply:
Oh okay, thanks for explaining! 

It looks like `@carbon/charts` generally has a peer dependency on `d3` (https://github.com/carbon-design-system/carbon-charts/blob/21ef21e2bb66e26cc00a97a5e03286ed97481119/packages/core/package.json#L61).
@FlorianRappl Do we need to add that dependency to `esm-styleguide` or should we do things like in this PR, i.e. reference `d3` in the different MFs when required?

Reply:
I think now its time to reveal and use the final weapon. Adding shared dependencies from individual MFs.

The idea is that one MF can bring in a dependency (presented in another chunk) which can be reused by another MF - without knowing each other. This way, the bundle sizes stay small and the performance is better, without losing independence or needing to know version numbers.

Based on https://docs.piral.io/guidelines/tutorials/15-share-dependencies#sharing-from-pilets.

Reply:
Great idea, yes.

Seems like we should build support for that into core. In which case, maybe we should merge this ticket for now so that it doesn't get stale, and then improve the dependency situation later.

Reply:
+1 on merging now. @donaldkibet, can you try removing this dependency and just see whether that breaks the pipeline? The other PRs also run through fine despite the warning. Then we could just merge it without adding this dep.